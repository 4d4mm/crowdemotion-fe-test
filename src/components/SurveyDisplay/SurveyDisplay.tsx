import React from 'react';
import { Link } from 'react-router-dom';
import { SurveyResponseType } from '../../types/survey';
import PageTitle from '../../components/PageTitle/PageTitle';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import Chip from '@material-ui/core/Chip';
import Paper from '@material-ui/core/Paper';
import { makeStyles } from '@material-ui/core/styles';
import Button from '../Button/Button';

const useStyles = makeStyles(theme=> ({
    paper: {
        padding: theme.spacing(2)
    },
    subtitle: {
        marginTop: theme.spacing(3),
        marginBottom: theme.spacing(2),
    }
}), {
   name: "SurveyDisplay"
})

const SurveyDisplay:React.SFC<SurveyResponseType> = ({
    author,
    category,
    createdAt,
    devices,
    exclusions,
    extProjectId,
    jobNumber,
    lineItems,
    notificationEmails,
    state,
    stateLastUpdatedAt,
    title,
    updatedAt,
}) => {
    const classes = useStyles({})
    return (
    <div>
        <PageTitle title={`${title} (Project)`} >
            <Link to="/">
                <Button variant="secondary">Back to Project List</Button>
            </Link>
        </PageTitle>
    <Grid container direction="column" spacing={3}>
    <Paper className={classes.paper}>
        
        <Grid container component="dl" spacing={3}>
            <Grid item xs={6} component="dt">
                <Typography variant="body1">Project ID</Typography>
            </Grid>
            <Grid item xs={6} component="dd">
                {extProjectId}
            </Grid>
            <Grid item xs={6} component="dt">
                <Typography variant="body1">Job Number</Typography>
            </Grid>
            <Grid item xs={6} component="dd">
                {jobNumber}
            </Grid>
            <Grid item xs={6} component="dt">
                <Typography variant="body1">Author</Typography>
            </Grid>
            <Grid item xs={6} component="dd">
                {author.name}
            </Grid>
            <Grid item xs={6} component="dt">
                <Typography variant="body1">Created At</Typography>
            </Grid>
            <Grid item xs={6} component="dd">
                {createdAt}
            </Grid>
            <Grid item xs={6} component="dt">
                <Typography variant="body1">Updated At</Typography>
            </Grid>
            <Grid item xs={6} component="dd">
                {updatedAt}
            </Grid>
            <Grid item xs={6} component="dt">
                <Typography variant="body1">Category</Typography>
            </Grid>
            <Grid item xs={6} component="dd">
                {category.surveyTopic.map((topic, index)=><Chip key={`topic_${index}`} label={topic} />)}
            </Grid>
            <Grid item xs={6} component="dd">
                <Typography variant="body1">Devices</Typography>
            </Grid>
            <Grid item xs={6} component="dd">
                {devices.map((device, index)=><Chip key={`device_${index}`} label={device} />)}
            </Grid>
            <Grid item xs={6} component="dt">
                <Typography variant="body1">State</Typography>
            </Grid>
            <Grid item xs={6} component="dd">
                <Chip label={state} />
            </Grid>
            <Grid item xs={6} component="dd">
                <Typography variant="body1">Exclusions</Typography>
            </Grid>
            <Grid item xs={6} component="dd">
                {exclusions.list.map((exclusion, index)=><Chip key={`exclusion_list_${index}`} label={exclusion} />)}
            </Grid>
            <Grid item xs={6} component="dd">
                <Typography variant="body1">Notification Emails</Typography>
            </Grid>
            <Grid item xs={6} component="dd">
                {notificationEmails.map((email, index)=><Chip key={`notification_email_${index}`} label={email} />)}
            </Grid>
        </Grid>
        
        </Paper>
        <Typography variant="h5" className={classes.subtitle}>Line Items</Typography>
        {lineItems.map((lineItem, lineItemIndex) => (
            <Paper key={`line_item_${lineItemIndex}`} className={classes.paper}>
                <Grid container component="dl" spacing={3}>
                <Grid item xs={6} component="dt">
                    <Typography variant="body1">Title</Typography>
                </Grid>
                <Grid item xs={6} component="dd">
                    <Chip label={lineItem.title} />
                </Grid>
                    <Grid item xs={6} component="dt">
                <Typography variant="body1">CountryISOCode</Typography>
                </Grid>
                <Grid item xs={6} component="dd">
                    <Chip label={lineItem.countryISOCode} />
                </Grid>
                    <Grid item xs={6} component="dt">
                <Typography variant="body1">Created At</Typography>
                </Grid>
                <Grid item xs={6} component="dd">
                    {lineItem.createdAt}
                </Grid>
                    <Grid item xs={6} component="dt">
                <Typography variant="body1">Days In Field</Typography>
                </Grid>
                <Grid item xs={6} component="dd">
                    <Chip label={lineItem.daysInField} />
                </Grid>
                    <Grid item xs={6} component="dt">
                <Typography variant="body1">Delivery Type</Typography>
                </Grid>
                <Grid item xs={6} component="dd">
                    <Chip label={lineItem.deliveryType} />
                </Grid>
                    
                    <Grid item xs={6} component="dt">
                <Typography variant="body1">Ext Line Item Id</Typography>
                </Grid>
                <Grid item xs={6} component="dd">
                    <Chip label={lineItem.extLineItemId} />
                </Grid>
                    <Grid item xs={6} component="dt">
                <Typography variant="body1">Indicative Incidence</Typography>
                </Grid>
                <Grid item xs={6} component="dd">
                    <Chip label={lineItem.indicativeIncidence} />
                </Grid>
                    <Grid item xs={6} component="dt">
                <Typography variant="body1">Ianguage ISO Code</Typography>
                </Grid>
                <Grid item xs={6} component="dd">
                    <Chip label={lineItem.languageISOCode} />
                </Grid>
                    <Grid item xs={6} component="dt">
                <Typography variant="body1">Launched At</Typography>
                </Grid>
                <Grid item xs={6} component="dd">
                    {lineItem.launchedAt}
                </Grid>
                    <Grid item xs={6} component="dt">
                <Typography variant="body1">Length Of Interview</Typography>
                </Grid>
                <Grid item xs={6} component="dd">
                    <Chip label={lineItem.lengthOfInterview} />
                </Grid>
                    
                    <Grid item xs={6} component="dt">
                <Typography variant="body1">Quota Plan</Typography>
                </Grid>
                
                <Grid item xs={6} component="dd">
                    <Chip label={lineItem.requiredCompletes} />
                </Grid>
                    
                    <Grid item xs={6} component="dt">
                <Typography variant="body1">State</Typography>
                </Grid>
                <Grid item xs={6} component="dd">
                    <Chip label={lineItem.state} />
                </Grid>
                    <Grid item xs={6} component="dt">
                <Typography variant="body1">State Last Updated At</Typography>
                </Grid>
                <Grid item xs={6} component="dd">
                    {lineItem.stateLastUpdatedAt}
                </Grid>
                    <Grid item xs={6} component="dt">
                <Typography variant="body1">Survey Test URL</Typography>
                </Grid>
                <Grid item xs={6} component="dd">
                    <Chip label={lineItem.surveyTestURL} />
                </Grid>
                    <Grid item xs={6} component="dt">
                <Typography variant="body1">Survey URL</Typography>
                </Grid>
                <Grid item xs={6} component="dd">
                    <Chip label={lineItem.surveyURL} />
                </Grid>
                    
                
                <Grid item xs={6} component="dt">
                    <Typography variant="body1">Updated At</Typography>
                </Grid>
                <Grid item xs={6} component="dd">
                    lineItem.updatedAt
                </Grid>
                    {/* <Grid item xs={6} component="dd">
                    <Chip label={lineItem.quotaPlan} />
                </Grid>
                    <Grid item xs={6} component="dt">
                <Typography variant="body1">requiredCompletes</Typography>
                </Grid>endLinks: Array<SurveyEndlink>,
                    targets: Array<SurveyTarget>,
                    sources: Array<SurveySource>; */}
                </Grid>
            </Paper>
        ))}
        </Grid>    
    </div>
)
}

export default SurveyDisplay;