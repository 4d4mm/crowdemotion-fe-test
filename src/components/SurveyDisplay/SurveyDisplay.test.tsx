import React from 'react';
import renderer from 'react-test-renderer';
import SurveyDisplay from './SurveyDisplay'
import { MemoryRouter, Route } from "react-router-dom";

const SurveyDisplayTester = () =>  (
    <SurveyDisplay
    {...{
        "author": {
        "name": "crowd_emotion_sample_api",
        "type": "api",
        "username": "crowd_emotion_sample_api"
        },
        "category": {
        "surveyTopic": [
            "AUTOMOTIVE",
            "BUSINESS"
        ]
        },
        "createdAt": "2019/11/07 21:12:02",
        "devices": [
        "mobile",
        "desktop",
        "tablet"
        ],
        "exclusions": {
        "list": [],
        "type": ""
        },
        "extProjectId": "project001",
        "jobNumber": "PO-1234",
        "lineItems": [
        {
            "countryISOCode": "US",
            "createdAt": "2019/11/07 21:12:02",
            "daysInField": 20,
            "deliveryType": "BALANCED",
            "endLinks": {
            "complete": "https://api.uat.pe.dynata.com/respondent/exit?rst=1&psid={psid}&med={calculatedSecurityCode}",
            "overquota": "https://api.uat.pe.dynata.com/respondent/exit?rst=3&psid={psid}",
            "screenout": "https://api.uat.pe.dynata.com/respondent/exit?rst=2&psid={psid}",
            "securityKey1": "93896",
            "securityKey2": "42614",
            "securityLevel": "MEDIUM"
            },
            "extLineItemId": "lineItem001",
            "indicativeIncidence": 20,
            "languageISOCode": "en",
            "launchedAt": "",
            "lengthOfInterview": 10,
            "quotaPlan": {
            "filters": [
                {
                "attributeId": "11",
                "operator": "INCLUDE",
                "options": [
                    "2"
                ]
                }
            ]
            },
            "requiredCompletes": 800,
            "sources": [
            {
                "id": 100
            }
            ],
            "state": "PROVISIONED",
            "stateLastUpdatedAt": "2019/11/07 21:12:02",
            "surveyTestURL": "www.mysurvey.com/test/survey?pid=<#DubKnowledge[1500/Entity id]>&k2=<#Project[Secure Key 2]>&psid=<#IdParameter[Value]>",
            "surveyURL": "www.mysurvey.com/live/survey?pid=<#DubKnowledge[1500/Entity id]>&k2=<#Project[Secure Key 2]>&psid=<#IdParameter[Value]>",
            "targets": [
            {
                "count": 800,
                "dailyLimit": 0,
                "softLaunch": 0,
                "type": "COMPLETE"
            }
            ],
            "title": "US College",
            "updatedAt": "2019/11/15 12:15:31"
        }
        ],
        "notificationEmails": [
        "api-test@dynata.com"
        ],
        "state": "PROVISIONED",
        "stateLastUpdatedAt": "2019/11/07 21:12:01",
        "title": "..",
        "updatedAt": "2019/11/15 12:15:31"
    }}
    />
);

describe('SurveyDisplay', () => {
    it('renders demo survey', () => {
        const renderedSurveyDisplay = renderer.create(
            <MemoryRouter keyLength={0} initialEntries={[{pathname: '/'}]}>
                <Route path="/" component={SurveyDisplayTester} />
            </MemoryRouter>
        ).toJSON();
        expect(renderedSurveyDisplay).toMatchSnapshot();
    
    })    
})