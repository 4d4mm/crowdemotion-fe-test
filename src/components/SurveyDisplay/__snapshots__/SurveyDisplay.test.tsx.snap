// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyDisplay renders demo survey 1`] = `
<div>
  <div
    className="MuiGrid-root PageTitle-main-3 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <h1
        className="MuiTypography-root MuiTypography-h3"
      >
        .. (Project)
      </h1>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          className="btn btn__secondary"
        >
          Back to Project List
        </button>
      </a>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation1 SurveyDisplay-paper-1 MuiPaper-rounded"
    >
      <dl
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Project ID
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          project001
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Job Number
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          PO-1234
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Author
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          crowd_emotion_sample_api
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Created At
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          2019/11/07 21:12:02
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Updated At
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          2019/11/15 12:15:31
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Category
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiChip-root"
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              AUTOMOTIVE
            </span>
          </div>
          <div
            className="MuiChip-root"
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              BUSINESS
            </span>
          </div>
        </dd>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Devices
          </p>
        </dd>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiChip-root"
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              mobile
            </span>
          </div>
          <div
            className="MuiChip-root"
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              desktop
            </span>
          </div>
          <div
            className="MuiChip-root"
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              tablet
            </span>
          </div>
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            State
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiChip-root"
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              PROVISIONED
            </span>
          </div>
        </dd>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Exclusions
          </p>
        </dd>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        />
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Notification Emails
          </p>
        </dd>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiChip-root"
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              api-test@dynata.com
            </span>
          </div>
        </dd>
      </dl>
    </div>
    <h5
      className="MuiTypography-root SurveyDisplay-subtitle-2 MuiTypography-h5"
    >
      Line Items
    </h5>
    <div
      className="MuiPaper-root MuiPaper-elevation1 SurveyDisplay-paper-1 MuiPaper-rounded"
    >
      <dl
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Title
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiChip-root"
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              US College
            </span>
          </div>
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            CountryISOCode
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiChip-root"
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              US
            </span>
          </div>
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Created At
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          2019/11/07 21:12:02
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Days In Field
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiChip-root"
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              20
            </span>
          </div>
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Delivery Type
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiChip-root"
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              BALANCED
            </span>
          </div>
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Ext Line Item Id
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiChip-root"
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              lineItem001
            </span>
          </div>
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Indicative Incidence
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiChip-root"
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              20
            </span>
          </div>
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Ianguage ISO Code
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiChip-root"
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              en
            </span>
          </div>
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Launched At
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Length Of Interview
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiChip-root"
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              10
            </span>
          </div>
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Quota Plan
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiChip-root"
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              800
            </span>
          </div>
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            State
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiChip-root"
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              PROVISIONED
            </span>
          </div>
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            State Last Updated At
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          2019/11/07 21:12:02
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Survey Test URL
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiChip-root"
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              www.mysurvey.com/test/survey?pid=&lt;#DubKnowledge[1500/Entity id]&gt;&k2=&lt;#Project[Secure Key 2]&gt;&psid=&lt;#IdParameter[Value]&gt;
            </span>
          </div>
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Survey URL
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiChip-root"
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              www.mysurvey.com/live/survey?pid=&lt;#DubKnowledge[1500/Entity id]&gt;&k2=&lt;#Project[Secure Key 2]&gt;&psid=&lt;#IdParameter[Value]&gt;
            </span>
          </div>
        </dd>
        <dt
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Updated At
          </p>
        </dt>
        <dd
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          lineItem.updatedAt
        </dd>
      </dl>
    </div>
  </div>
</div>
`;
